src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -81,6 +98,50 @@ public void testParseDeleteChange() throws Exception {
     }
 
     @Test
+    public void testParseRenameChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorldFoo");
+            GitUtils.removeFileAndCommit(repository, "HelloWorld");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
+                    .path("HelloWorldFoo")
+                    .build();
+            assertEquals("The change is not as expected", expected, changes.get(0));
+        }
+    }
+
+    @Test
+    public void testParseCopyChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "a.txt");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "b.txt");
+            GitUtils.addAnEmptyFileAndCommit(repository, "src/com/foo/c.java");
+            GitUtils.addAnEmptyFileAndCommit(repository, "src/com/foo/d.java");
+            GitUtils.removeFileAndCommit(repository, "a.txt");
```

Comment:
When I debug `DiffParser` everything in `DiffEntry` is recognized as `ADD` for me.
Is this just because we don't turn on recognizing copies?
When I open the repsository in EGit, and squash the commits, it does say 
copy' and 'rename'.
If I do remove, `Delete a.txt` then EGit says there is no `copies`.
It is weird behavior.

Let's ignore this for now.